terraform/azure/infra/main.tf

```diff
@@ -151,14 +151,43 @@ resource "azurerm_web_application_firewall_policy" "wandb" {
   }
 }
 
+resource "azurerm_network_security_group" "wandb" {
+  # The AKS Application Gateway ingress controller is only able to use a private IP
+  # if the gateway is using Standard_v2 or WAF_v2.  They both currently require
+  # a public IP, see: https://github.com/Azure/application-gateway-kubernetes-ingress/issues/741
+  # This rule blocks all internet access to the gateway and is only used if
+  # var.deployment_is_private
+  name                = "wandb-private-deployment-security-group"
+  location            = azurerm_resource_group.wandb.location
+  resource_group_name = azurerm_resource_group.wandb.name
+
+  security_rule {
```

Comment:
Do we need to add any additional rules here? https://docs.microsoft.com/en-us/azure/application-gateway/configuration-infrastructure#allow-access-to-a-few-source-ips

Reply:
We could make the generic network policy block IP's but that's already possible / implemented with the WAF configuration.  Definitely something we could revisit.